Makefile

```diff
@@ -143,16 +143,17 @@ TF_CFLAGS_aarch64	=	-mgeneral-regs-only -mstrict-align
 ASFLAGS_aarch32		=	-march=armv8-a
 TF_CFLAGS_aarch32	=	-march=armv8-a
 
-ASFLAGS			+= 	-nostdinc -ffreestanding -Wa,--fatal-warnings	\
-				-Werror -Wmissing-include-dirs			\
-				-D__ASSEMBLY__ $(ASFLAGS_$(ARCH))		\
-				${DEFINES} ${INCLUDES}
-TF_CFLAGS		+= 	-nostdinc -ffreestanding -Wall			\
-				-Werror -Wmissing-include-dirs			\
-				-std=c99 -c -Os					\
-				$(TF_CFLAGS_$(ARCH))				\
-				${DEFINES} ${INCLUDES}
-TF_CFLAGS		+=	-ffunction-sections -fdata-sections
+# common to CPP, CC, AS
+COMMON_FLAGS		=	${DEFINES} ${INCLUDES} -nostdinc		\
+				-Wmissing-include-dirs -Werror
+
+CPPFLAGS		+=	$(COMMON_FLAGS) -D__ASSEMBLY__
```

Comment:
I'm a bit uncomfortable with -D__ASSEMBLY__ being defined in CPPFLAGS here... Potentially, CPPFLAGS could be used to pre-process any file, asm files as well as C files (although it seems to be used only for linker scripts at the moment). For this reason, I'd rather have only the ${COMMON_FLAGS} in the CPPFLAGS, and have the linker script pre-processing rule explictly pass the -D__ASSEMBLY__ flag in addition to the CPPFLAGS.